mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1169,43 +1169,123 @@ def _log_eval_table(self):
             )
         else:
             data = self.dataset.features_data.assign(outputs=self.y_pred)
+        data = data.assign(**self.metrics_dict)
         mlflow.log_table(data, artifact_file=f"{metric_prefix}{_EVAL_TABLE_FILE_NAME}")
 
-    def _evaluate_question_answering(self):
-        self._log_eval_table()
-        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
-        self.artifacts[name] = JsonEvaluationArtifact(
-            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
+    def _calculate_perplexity(self, predictions):
+        try:
+            import evaluate
+
+            perplexity = evaluate.load("perplexity", module_type="metric")
+        except Exception as e:
+            _logger.warning(
+                f"Failed to load 'perplexity' metric (error: {e!r}), skipping metric logging."
+            )
+            return
+
+        results = perplexity.compute(predictions=predictions, model_id="gpt2")
```

Comment:
Is this the recommended model for perplexity? GPT-2 is pretty old at this point

Reply:
The metric card for perplexity on huggingface uses gpt2. Not sure what would be a better choice here.

Reply:
Other options are listed here:

https://huggingface.co/docs/transformers/main/en/model_doc/auto#transformers.AutoModelForCausalLM